### PR TITLE
Bugfix TypeError in generate_report when using NiftiMasker for mask_img

### LINKED ISSUE #3034: Method generate_report raises TypeError when mask_img of FirstLevelModel is NiftiMasker
`FirstLevelModel` accepts `mask_img` to be a `NiftiMasker`. However, calling `generate_report()` on such an instance will raise a `TypeError`, due to the fact that `_mask_to_svg()` calls `plot_roi()` on attribute `mask_img` (`plot_roi()` would need `mask_img` to be a `NiftiImage`, not a `NiftiMasker`).

Nilearn version: 0.8.2.dev

### Expected behavior
The report should be generated flawlessly.

### Actual behavior
An explicit `TypeError` is raised, but users can't easily fix this bug as it lies in unexposed methods.

### Steps and code to reproduce bug
Here is an incomplete code snippet leading to this bug:

```python
masker = NiftiMasker()
masker.fit(anatomical_img)

glm = FirstLevelModel(t_r=t_r, mask_img=masker)
glm.fit(functional_imgs, events)
glm.generate_report(["condition_name"])
```

### Fix
A simple fix would be to call `_mask_to_svg()` with `NiftiImage` parameters only. One simply need to modify the following lines to do a proper type check: https://github.com/nilearn/nilearn/blob/80c55fc43d614c4edf35c42186ac414ac3b770e3/nilearn/reporting/glm_reporter.py#L217-L220

I can take care of this soon!

### PR DESCRIPTION
**Pull request by alexisthual**
Closes #3034.

Changes proposed in this pull request:

- add test failing if bugfix is not implemented
- do simple type check of attribute `mask_img` of `FirstLevelModel` when generating GLM model

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: doc/whats_new.rst [hunk 1]
```diff
@@ -33,6 +33,11 @@ Fixes
   as an array.
   (See issue `#3012 <https://github.com/nilearn/nilearn/issues/3012>`_) and fix
   `#3013 <https://github.com/nilearn/nilearn/pull/3013>`_).
+- :func:`nilearn.glm.first_level.FirstLevelModel.generate_report` threw a `TypeError`
```
Comment (NicolasGensollen) [inline doc/whats_new.rst:36]:
```suggestion
- :meth:`nilearn.glm.first_level.FirstLevelModel.generate_report` threw a `TypeError`
```

File: doc/whats_new.rst [hunk 2]
```diff
@@ -33,6 +33,11 @@ Fixes
...
+  when `FirstLevelModel` was instantiated with `mask_img` being a `NiftiMasker`.
```
Comment (NicolasGensollen) [inline doc/whats_new.rst:37]:
```suggestion
  when `FirstLevelModel` was instantiated with `mask_img` being a :class:`~nilearn.input_data.NiftiMasker`.
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @alexisthual !
Could you rebase/merge master to get fix #3060 and run a full build?

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @alexisthual !

**Pull request merged**